File: app_codebase/python_base_05/bin/core/auth/verify_access.py
"""Shared access JWT verification for HTTP guards and WS auth handshake."""

from __future__ import annotations

from jwt.exceptions import ExpiredSignatureError, InvalidTokenError

from core.auth import token_service
from core.auth.contracts.auth_context_contract import AuthContext
from core.errors.app_error import AppError
from core.errors.error_codes import INVALID_TOKEN, TOKEN_EXPIRED, UNAUTHORIZED


def _verify_access_token(token: str) -> AuthContext:
    try:
        return token_service.verify_access(token.strip())
    except ExpiredSignatureError as err:
        raise AppError(TOKEN_EXPIRED) from err
    except InvalidTokenError as err:
        raise AppError(INVALID_TOKEN) from err


def verify_bearer_or_raise(token: str | None) -> AuthContext:
    if not token or not str(token).strip():
        raise AppError(UNAUTHORIZED, message="Bearer token required")
    return _verify_access_token(token)


def verify_access_or_raise(token: str | None) -> AuthContext:
    if not token or not str(token).strip():
        raise AppError(UNAUTHORIZED, message="access_token required")
    return _verify_access_token(token)

File: app_codebase/dart_bkend_base_02/bin/core/auth/verify_access.dart
/// Shared access JWT verification for HTTP guards and WS auth handshake.
library;

import 'package:dart_jsonwebtoken/dart_jsonwebtoken.dart';

import '../errors/app_error.dart';
import '../errors/error_codes.dart';
import 'auth_context.dart';
import 'jwt_token_service.dart';

AuthContext _verifyAccessToken(String token) {
  try {
    return tokenService.verifyAccess(token.trim());
  } on JWTExpiredException {
    throw AppError(tokenExpired);
  } on JWTException {
    throw AppError(invalidToken);
  }
}

AuthContext verifyBearerOrThrow(String? token) {
  if (token == null || token.trim().isEmpty) {
    throw AppError(unauthorized, message: 'Bearer token required');
  }
  return _verifyAccessToken(token);
}

AuthContext verifyAccessOrThrow(String? token) {
  if (token == null || token.trim().isEmpty) {
    throw AppError(unauthorized, message: 'access_token required');
  }
  return _verifyAccessToken(token);
}
